src/responder.ts

```diff
@@ -235,47 +347,50 @@ export class EthereumDedicatedResponder extends EthereumResponder {
 
 export class EthereumResponderManager {
     private responders: Set<EthereumResponder> = new Set();
+    private gasPolicy: GasPolicy;
 
     constructor(private readonly signer: ethers.Signer) {
-
+        this.gasPolicy = new DoublingGasPolicy(this.signer.provider);
     }
 
     public respond(appointment: IEthereumAppointment) {
-        const ethereumResponse = appointment.getResponse();
+        const ethereumResponseData = appointment.getResponseData();
 
-        const responder = new EthereumDedicatedResponder(this.signer, appointment.id, ethereumResponse, 10);
+        const responder = new EthereumDedicatedResponder(this.signer, this.gasPolicy, 40, 10);
```

Comment:
The lock inside the responder stops multiple calls to respond on a given responder, but it doesnt stop multiple responders being created with the same key which amounts to the same thing. I would create a KeyStore and use it to try and obtain a key before creating your responder.

Reply:
Shouldn't this be part of the ResponderManager? My current plan is that the ResponderManager handles the pool of wallets and allocates them to the responder implementations, while the Responder assumes that it has exclusive control of the wallet (as I mention in the class documentation).

Reply:
Yep, the responder manager can call the key store. The key store should be a separate object though.